src/proc.cr

```diff
@@ -35,6 +35,10 @@ struct Proc
     call(other)
   end
 
+  def hash
+    object_id.hash
+  end
```

Comment:
We could make Value's hash be computed from the raw bytes, but maybe it's better to ensure that each subclass defines a correct logic.